### PR TITLE
Checkbox test

### PR DESCRIPTION
**Pull request by williamdaconceicao**
```
 TypeError: Cannot read property 'indeterminate' of null

 export default class Checkbox extends Component {
  componentDidMount() {
    if ((this.inputRef.indeterminate !== null ) || (this.props.indeterminate !== null)){
      this.inputRef.indeterminate = this.props.indeterminate;
   }
  } 
```

Here it's the error that I got

### CODE REVIEW

File: tests/demos/checkboxes.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Checkboxes from "@demos/components/checkboxes";
+
+describe("demos/components/checkboxes", () => {
+  it("renders itself correctly", () => {
+    const component = renderer.create(<Checkboxes />)
+    console.log(component)
```
Comment (willdurand) [inline tests/demos/checkboxes.test.js]: so this is not a proper test.

I guess it does not work because `inputRef` is not defined. I think you have to "mock" it like here: https://github.com/Zoapp/react-materialcomponents/blob/c29e019176b9ab5236db07b39022677cc164f836/tests/dialog/dialog.test.js#L29.

I'd say something like this:

```
const createNodeMock = () => ({
    indeterminate: null,
});

const component = renderer.create(
    <Checkboxes />,
    { createNodeMock }
);
```

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
-    "test": "jest",
+    "test": "yarn run lint && yarn run jest",
```
Comment (willdurand) [inline package.json]: you should revert this change

File: tests/demos/checkboxes.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import Checkboxes from "@demos/components/checkboxes";
+
+describe("demos/components/checkboxes", () => {
+  it("renders itself correctly", () => {
+    const component = renderer.create(
+    	<Checkboxes />,
+    	{ createNodeMock }).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+ const createNodeMock = () => ({
```
Comment (willdurand) [inline tests/demos/checkboxes.test.js]: put this first, because you are using it *before*

**Comment by varunrajasekhar** [comment]
I am having a weird issue:
I have a button called assign and when I click on it, a modal with userslist(checkboxes) opens
This modal has a confirm and cancel button. THe buttons functionality should be such as
When I click on confirm, the checked or unchecked data(no matter what selection is made) should save and when I click on cancel any data should revert back to confirmed(saved) data.
I have built my component such that when loading initially, I am copying the state and then the 'copystate' changes are saved to original state when confirmed and when cancelled, original state changes are copied to 'copystate'.
But what is happening is the state is getting updated no matter what change I make in the copystate. and the copystate and original state are having the same data.

Can somebody help me with this:
Below is my render method:

render() {
    const ready = this.state.ready;
    return (
      <div>
        {ready &&
          this.state.data.addOnProducts.map((val, idx) => {
            const userListIndex = val.indexOf()
            return (
              <div key={idx}>
                <button className={`btn btn--rounded btn--assign`} onClick={() => this.openModal(idx)}>
                  {'assign'}
                </button>
                {this.state.showModal && (
                 
                  <CheckboxList
                    key={idx}
                    list={this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList}
                    selectHandler={this.assignHandler}
                    //label={this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList}
                  />
                )}
              </div>
            );
          })}
      </div>
    );
  }

Below is my assign handler:

assignHandler(idx) {
    const newData = Object.assign({}, this.state.newData);
    console.log('state list',this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);
    console.log('newData list',this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);

    newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList.map((user, index) => {
      if (index === idx) {
        user.selected = !user.selected;
      }

      return user;
    });

    this.setState(
    {
      newData: Object.assign({}, newData)
    }, function() {
      console.log('state list',this.state.data.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);
      console.log('newData list',this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);
    });
  };
This is Cancel Method:

 cancelAssignUsers() {
    
        const stateUsersList = Object.assign({}, this.state.data.addOnProducts[this.state.currentAddOnIndex].usersData);
        const newDataCopy = Object.assign({}, this.state.newData);
    console.log(stateUsersList.usersList);
        newDataCopy.addOnProducts[this.state.currentAddOnIndex].usersData.usersList = stateUsersList.usersList;
    console.log(newDataCopy.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);
        this.setState({
          showModal: false,
          newData: Object.assign({}, newDataCopy)
        });
  }

this is my confirm method:
confirmAssignUsers() {
    
    const newDataUsersList = Object.assign(
      {},
      this.state.newData.addOnProducts[this.state.currentAddOnIndex].usersData
    );
    const stateCopy = Object.assign({}, this.state.data);

    stateCopy.addOnProducts[this.state.currentAddOnIndex].usersData.usersList = newDataUsersList.usersList;
console.log(stateCopy.addOnProducts[this.state.currentAddOnIndex].usersData.usersList);
    this.setState({
      showModal: false,
      data: Object.assign({}, stateCopy)
    });
  }

File: package.json [hunk 4]
```diff
@@ -11,7 +11,8 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
-    "test": "yarn run lint && yarn run jest",
+    "check": "yarn run lint && yarn run jest",
```
Comment (willdurand) [inline package.json]: :+1:

**Comment by willdurand** [comment]
@varunrajasekhar please open an issue for that.

File: package.json [hunk 5]
```diff
@@ -11,7 +11,8 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
-    "test": "jest",
+    "check": "yarn run lint && yarn run jest",
+    "test": "yarn run jest",
```
Comment (willdurand) [inline package.json]: you should not change this line, it is indeed `"test": "jest"`

File: package.json [hunk 6]
```diff
@@ -11,7 +11,8 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
-    "test": "jest",
+    "check": "yarn run lint && yarn run jest",
```
Comment (willdurand) [inline package.json]: it should be `yarn lint && yarn test`

File: package.json [hunk 7]
```diff
@@ -11,8 +11,8 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
-    "check": "yarn run lint && yarn run jest",
-    "test": "yarn run jest",
+    "check": "yarn lint && yarn jest",
```
Comment (willdurand) [inline package.json]: I insist, it should be `yarn test` not `yarn jest`

Comment (williamdaconceicao) [reply package.json]: Sorry haven't see it ;)

File: package.json [hunk 8]
```diff
@@ -11,6 +11,7 @@
     "compile:css": "yarn clean-dist && cleancss -o dist/compressed.css node_modules/material-components-web/dist/material-components-web.css node_modules/dialog-polyfill/dialog-polyfill.css src/libs/css/app.css",
     "dev": "yarn compile:dev && yarn start:dev",
     "lint": "eslint src/libs --ext=js --ext=jsx && eslint src/demos --ext=js --ext=jsx",
+    "check": "yarn lint && yarn jest",
```
Comment (willdurand) [inline package.json]: I insist, it should be `yarn test` not `yarn jest`
